aten/src/ATen/core/jit_type.h

```diff
@@ -1216,7 +1216,9 @@ CAFFE2_API c10::optional<TypePtr> unifyTypes(
     const TypePtr& t1,
     const TypePtr& t2);
 
-CAFFE2_API c10::optional<TypePtr> unifyTypeList(at::ArrayRef<TypePtr> elements);
+CAFFE2_API c10::optional<TypePtr> unifyTypeList(
+    at::ArrayRef<TypePtr> elements,
+    std::ostream& why_not);
```

Comment:
Maybe make the `ostream` optional here ? `unifyTypes` doesn't have the same mandatory `ostream` argument.

Reply:
`unifyTypes` doesn't need one since you know what happened, while in `unifyTypeList` there needs to be some way to bubble up the reason since it could have happened to any element in the list